src/google/protobuf/compiler/importer.cc

```diff
@@ -490,6 +490,19 @@ io::ZeroCopyInputStream* DiskSourceTree::OpenVirtualFile(
 
 io::ZeroCopyInputStream* DiskSourceTree::OpenDiskFile(
     const std::string& filename) {
+  struct stat sb;
+  int ret = 0;
+  do {
+    ret = stat(filename.c_str(), &sb);
+  } while (ret != 0 && errno == EINTR);
+#ifdef _WIN32
```

Comment:
@junekhan Sorry for taking so long to get back to this, but I have just one comment. I was thinking you might be able to check whether the path is a directory by doing something like this:
`if (sb.st_mode & S_IFDIR)`
I believe that should work the same way on Linux, Mac, and Windows. Also that should allow things to work even with files like `/dev/stdin`, which could be useful even though they're not regular files.

Reply:
Good to know this universal statement. Thanks.
I checked the references of the function and was not sure of whether it's correct to open files other than regular files. But now I see your point and know it's useful.